llvm/lib/Target/LoongArch/LoongArchOptWInstrs.cpp

```diff
@@ -167,18 +166,45 @@ static bool hasAllNBitUsers(const MachineInstr &OrigMI,
       case LoongArch::MOVGR2FCSR:
       case LoongArch::MOVGR2FRH_W:
       case LoongArch::MOVGR2FR_W_64:
+      case LoongArch::VINSGR2VR_W:
```

Comment:
Is it possible to add some tests for these instructions?

Reply:
I've added test cases for the `vreplgr2vr.w` instruction. For vector instructions, effective optimizations often arise in cases involving complex multi-level data flow relationships.  I plan to add more test cases in the future if I identify opportunities for further simplification.